workers/unity/Packages/io.improbable.gdk.core/Utility/Snapshot.cs

```diff
@@ -23,11 +34,21 @@ public class Snapshot
         /// <returns>The entity ID assigned to the entity in the snapshot.</returns>
         public EntityId AddEntity(EntityTemplate entityTemplate)
         {
-            var entityId = new EntityId(entities.Count + 1);
-            entities[entityId] = entityTemplate.GetEntity();
+            var entityId = GetNextEntityId();
+            AddEntity(entityId, entityTemplate);
             return entityId;
         }
 
+        /// <summary>
+        ///     Adds an entity to the snapshot
+        /// </summary>
+        /// <param name="entityId">The entity ID of the entity to be added to the snapshot</param>
+        /// <param name="entityTemplate">The entity to be added to the snapshot.</param>
+        public void AddEntity(EntityId entityId, EntityTemplate entityTemplate)
```

Comment:
Add a remark that this invalidates `GetNextEntityId`

Reply:
what do you mean "invalidates"?

Reply:
If I add an entity with hardcoded ID 1 here, it might still be given by the function `GetNextEntityId`

Reply:
surely users would only call this method after getting an id from `GetNextEntityId`?

Reply:
But there is nothing guaranteeing that, and I can totally see them not do this.